src/main/java/medicaldoctor/util/EmailNameGenerator.java

```diff
@@ -0,0 +1,10 @@
+package medicaldoctor.util;
+
+public class EmailNameGenerator implements EmailGenerator{
+    
+    @Override
+    public String generateEmail(String first_name, String last_name) {
+        return first_name.substring(0, 1) + last_name.substring(1, last_name.length()) + "@MCARE.com";
```

Comment:
how about just passing in the username.
also add email generator to createNewUser service